pkgs/stdenv/cross/default.nix

```diff
@@ -26,19 +26,24 @@ in lib.init bootStages ++ [
       assert vanillaPackages.stdenv.buildPlatform == localSystem;
       assert vanillaPackages.stdenv.hostPlatform == localSystem;
       assert vanillaPackages.stdenv.targetPlatform == localSystem;
-      vanillaPackages.stdenv.override { targetPlatform = crossSystem; };
+      vanillaPackages.stdenv.override {
+        targetPlatform = if crossSystem == null
+                         then localSystem
+                         else crossSystem;
```

Comment:
I think would be better to in `pkgs/top-level/default.nix` default `crossSystem` to be `localSystem`. Then all these can stay the same.